### DISCUSSION TITLE
Does it work when laravel spatie/laravel-permission is installed already

### DISCUSSION THREAD

**Discussion by raliqala** [shopperlabs/framework#191] [Q&A]
Hi.  
I have an existing laravel application that uses spatie/laravel-permission.  
I wanted to know before I install if it works with spatie/laravel-permission

**Comment by mckenziearts** [comment]
Hey @raliqala, yes, fully compatible.

Shopper uses `spatie/laravel-permission` internally for its roles and permissions system, so there's nothing extra to integrate. If you already have it installed, Shopper piggybacks on the same `roles` and `permissions` tables and adds its own permissions during install. Just make sure both sets of migrations have run and the `HasRoles` trait is on your `User` model (the installer handles that part).
